mlrun/model_monitoring/db/tsdb/v3io/v3io_connector.py

```diff
@@ -1561,91 +1562,115 @@ def get_drift_data(
         if not raw_frames:
             return mm_schemas.ModelEndpointDriftValues(values=[])
 
-        aggregated_data = self._aggregate_raw_drift_data(
+        # Combine aggregation, filtering, and conversion in one pass
+        drift_values = self._process_drifted_endpoints_data(
             raw_frames=raw_frames, start=start, end=end, interval=interval
         )
-        if not aggregated_data:
-            return mm_schemas.ModelEndpointDriftValues(values=[])
-
-        # Filter to only include entries with max result_status >= 1
-        filtered_data = [
-            (endpoint_id, timestamp, max_status)
-            for endpoint_id, timestamp, max_status in aggregated_data
-            if max_status >= 1
-        ]
 
-        if not filtered_data:
-            return mm_schemas.ModelEndpointDriftValues(values=[])
-
-        return self._convert_drift_data_to_values(aggregated_data=filtered_data)
+        return drift_values
 
     @staticmethod
-    def _aggregate_raw_drift_data(
+    def _process_drifted_endpoints_data(
         raw_frames: list[v3io_frames.client.RawFrame],
         start: datetime,
         end: datetime,
         interval: str,
-    ) -> list[tuple[str, datetime, float]]:
+    ) -> mm_schemas.ModelEndpointDriftValues:
         """
-        Aggregate raw drift data from RawFrame objects.
+        Optimized single-pass processing of drift data from RawFrame objects.
+        Combines aggregation, filtering, and conversion into one operation.
 
         :param raw_frames: List of RawFrame objects containing drift data.
         :param start:      Start datetime for filtering data.
         :param end:        End datetime for filtering data.
         :param interval:   Time interval string (e.g., '5min') for aggregation
 
-        :returns: list of tuples: (endpoint_id, timestamp, max_result_status)
+        :returns: ModelEndpointDriftValues with counts of suspected and detected per timestamp
         """
+
         if not raw_frames:
-            return []
+            return mm_schemas.ModelEndpointDriftValues(values=[])
 
-        # Parse interval to get timedelta
+        # Pre-compute constants
         interval_td = pd.Timedelta(interval)
+        interval_ns = interval_td.value  # nanoseconds for integer arithmetic
+        start_ns = pd.Timestamp(start).value
+        end_ns = pd.Timestamp(end).value
+
+        suspected_val = mm_schemas.constants.ResultStatusApp.potential_detection.value
+        detected_val = mm_schemas.constants.ResultStatusApp.detected.value
+
+        # Single dictionary to track: bucket_start_ns -> {endpoint_id -> max_status}
+        # This allows us to calculate max per endpoint per bucket in one pass
+        bucket_endpoint_status = {}
+
+        # Single pass through all frames and data points
+        total_data_points = 0
+        filtered_data_points = 0
```

Comment:
both vars are only assigned-to but never used - please check if can be removed?

Reply:
we do use them in https://github.com/mlrun/mlrun/blob/fbe6dd9ef3bff7941e2968cc3406bd69063a9092/mlrun/model_monitoring/db/tsdb/v3io/v3io_connector.py#L1615 and https://github.com/mlrun/mlrun/blob/fbe6dd9ef3bff7941e2968cc3406bd69063a9092/mlrun/model_monitoring/db/tsdb/v3io/v3io_connector.py#L1624

Reply:
These are also just assignments - it's not a case where these vars are **used**.